File: authorization-service/src/main/java/com/mazeltov/authorization/service/config/MySwaggerConfig.java
package com.mazeltov.authorization.service.config;

import org.springframework.beans.factory.annotation.*;
import org.springframework.context.annotation.*;
import springfox.documentation.builders.*;
import springfox.documentation.service.*;
import springfox.documentation.spi.*;
import springfox.documentation.spring.web.plugins.*;
import springfox.documentation.swagger2.annotations.*;

import java.util.*;
import java.util.function.*;

import static com.mazeltov.common.swagger.ConfigKt.*;
import static springfox.documentation.builders.PathSelectors.*;

@Configuration
@EnableSwagger2
public class MySwaggerConfig {
    @Bean
    public Docket api() {
        List<SecurityScheme> schemeList = new ArrayList<SecurityScheme>();
        schemeList.add(new ApiKey("JWT", "Authorization", "header"));
        return new Docket(DocumentationType.SWAGGER_2)
                .securityContexts(securityContext())
                .securitySchemes(schemeList)
                .select()
                .apis(RequestHandlerSelectors.any())
                .paths(paths())
                .build();
    }

    @Value("${api.authorization-service.rout}")
    private String authorizationService;


    private Predicate<String> paths() {
        return regex("/.*")
                .or(regex(authorizationService + ".*"));

    }
}

File: cart-service/src/main/kotlin/com/mazeltov/cart/service/config/MySwaggerConfig.kt
package com.mazeltov.cart.service.config

import com.mazeltov.common.swagger.*
import org.springframework.beans.factory.annotation.*
import org.springframework.context.annotation.*
import springfox.documentation.builders.*
import springfox.documentation.builders.PathSelectors.*
import springfox.documentation.service.*
import springfox.documentation.spi.*
import springfox.documentation.spring.web.plugins.*
import springfox.documentation.swagger2.annotations.*
import java.util.function.*
/**
 * Класс для настройки Swagger
 */
@Configuration
@EnableSwagger2
class MySwaggerConfig {
    @Bean
    fun api(): Docket {
        val schemeList: MutableList<SecurityScheme> = ArrayList<SecurityScheme>()
        schemeList.add(ApiKey("JWT", "Authorization", "header"))
        return Docket(DocumentationType.SWAGGER_2)
            .securityContexts(securityContext())
            .securitySchemes(schemeList)
            .select()
            .apis(RequestHandlerSelectors.any())
            .paths(paths())
            .build()
    }

    @Value("\${api.cart-service.rout}")
    private lateinit var cartService: String

    private fun paths(): Predicate<String> {
        return regex("/.*")
            .or(regex("$cartService.*"))
    }
}
